### PR TITLE
Tina stop detection

### PR DESCRIPTION
**Pull request by tina-ni**
Lachesis can now optionally refine its stop labels with `postprocessing="dbscan"`. It summarizes the initially detected stops, groups nearby stops into temporary locations, and merges consecutive stops assigned to the same location when the temporal gap is within `dt_max`. The temporary locations are discarded: the original pings receive the final consecutive stop labels, while noise remains `-1`. The per-user Lachesis wrappers apply this process independently to each user.

This also adds public tools for using the two postprocessing steps separately. `detect_locations` assigns integer location IDs with DBSCAN or a custom callable, retains DBSCAN noise as singleton locations, and can return location centers and extents. `merge_stops` combines a chronological single-user stop table by location and time gap, or processes location-labeled pings with the built-in grid-based method or a custom stop-detection callable.

The postprocessing is also integrated as part of the existing Lachesis APIs, by means of argument `postprocessing="dbscan"`

### CODE REVIEW

File: nomad/stop_detection/clustering.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+import numpy as np
+import pandas as pd
+from sklearn.cluster import DBSCAN
+
+import nomad.io.base as loader
+
+
+EARTH_RADIUS_METERS = 6_371_000
```
Comment (paco-barreras) [inline nomad/stop_detection/clustering.py]: There's hardcoded earth diameters and radii all over. It's time to move them to constants.py

File: nomad/stop_detection/clustering.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+
+def dbscan_labels(stops, epsilon=100, num_samples=1, traj_cols=None, **kwargs):
```
Comment (paco-barreras) [inline nomad/stop_detection/clustering.py]: Try to make it consistent with other functions that take stop tables as input.

File: nomad/stop_detection/clustering.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    Cluster stop locations with DBSCAN.
+
+    Parameters
+    ----------
+    stops : pd.DataFrame
+        Stop table containing x/y or longitude/latitude coordinates.
+    epsilon : float, default 100
+        Maximum distance between neighboring stops. Units match x/y coordinates
+        or are meters for longitude/latitude coordinates.
+    num_samples : int, default 1
```
Comment (paco-barreras) [inline nomad/stop_detection/clustering.py]: We should make it consistent with the density based algorithms, `min_pts`
